canopen_ros2_control/src/canopen_system.cpp

```diff
@@ -127,69 +125,37 @@ void CanopenSystem::spin() {
-                             tmp_master_bin))
-    {
-        auto node_map = device_container_->get_node_instance_wrapper_map();
-        RCLCPP_INFO(kLogger, "Number of nodes: '%zu'", node_map.size());
-
-        auto reg_dr = device_container_->get_registered_drivers();
-        RCLCPP_INFO(kLogger, "Number of registered drivers: '%zu'", reg_dr.size());
-        for(auto it = reg_dr.begin(); it != reg_dr.end(); it++){
-            auto proxy_driver =  std::static_pointer_cast<ros2_canopen::ProxyDriver>(device_container_->get_node(it->second.first));
-
-            auto nmt_state_cb = [&](canopen::NmtState nmt_state, uint8_t id){
-//                RCLCPP_INFO(
-//                        kLogger,
-//                        "Slave %u: Switched NMT state to %u",
-//                        id,
-//                        nmt_state);
-
-                // store into map
-                canopen_data_[id].nmt_state.set_state(nmt_state);
-            };
-            // register callback
-            proxy_driver->register_nmt_state_cb(nmt_state_cb);
-
-            auto rpdo_cb = [&](ros2_canopen::COData data, uint8_t id){
-//                RCLCPP_INFO(kLogger,
-//                        "slave: '%u',"
-//                        " data: '%u',"
-//                        " index: '%u', "
-//                        "subindex: '%u',"
-//                        " type: '%u',",
-//                        id, data.data_, data.index_, data.subindex_, data.type_);
-                canopen_data_[id].rpdo_data.set_data(data);
-            };
-            // register callback
-            proxy_driver->register_rpdo_cb(rpdo_cb);
-
-            RCLCPP_INFO(kLogger, "\nRegistered driver:\n    name: '%s'\n    node_id: '%u'\n    driver: '%s'", it->first.c_str(), it->second.first, it->second.second.c_str());
-
-        }
-
-        auto act_dr = device_container_->get_active_drivers();
-        RCLCPP_INFO(kLogger, "Number of active drivers: '%zu'", act_dr.size());
-        for(auto it = act_dr.begin(); it != act_dr.end(); it++){
-            RCLCPP_INFO(kLogger, "\nActive driver:\n    name: '%s'\n    node_id: '%u'\n    driver: '%s'", it->first.c_str(), it->second.first, it->second.second.c_str());
-        }
-        RCLCPP_INFO(device_container_->get_logger(), "Initialisation successful.");
-    }
-    else
-    {
-        RCLCPP_INFO(device_container_->get_logger(), "Initialisation failed.");
+  device_container_->init(info_.hardware_parameters["can_interface"],
+                            info_.hardware_parameters["master_config"],
+                            info_.hardware_parameters["bus_config"],
+                            tmp_master_bin);
+  auto drivers = device_container_->get_registered_drivers();
+  RCLCPP_INFO(kLogger, "Number of registered drivers: '%zu'", device_container_->count_drivers());
+  for (auto it = drivers.begin(); it != drivers.end(); it++){
+    auto proxy_driver = std::static_pointer_cast<ros2_canopen::ProxyDriver>(it->second);
```

Comment:
If we would want to treat ProxyDriver and Cia402Driver, we should use the node_canopen_interface.
```suggestion
    auto proxy_driver = std::static_pointer_cast<ros2_canopen::node_interfaces::node_canopen_proxy_driver>(it->second)->get_node_canopen_driver();
```